src/test/java/org/kiwiproject/eureka/EurekaServletHandlerTest.java

```diff
@@ -412,4 +415,93 @@ void shouldReturn500ForStatusOnTriggeredFailure() {
             assertInternalServerErrorResponse(response);
         }
     }
+
+    @Nested
+    class PostRegistration {
+
+        @Test
+        void shouldRegisterApplicationAndReturn204() {
+            var postBody = Map.of(
+                    "instance", Map.of(
+                            "hostName", "localhost",
+                            "homePageUrl", "http://localhost",
+                            "statusPageUrl", "http://localhost/status",
+                            "healthCheckUrl", "http://localhost/healthcheck",
+                            "status", "STARTING",
+                            "port", Map.of("$", 8080),
+                            "securePort", Map.of("$", 8081),
+                            "vipAddress", APP_NAME
+                    )
+            );
+
+            var response = client.target(server.getURI())
+                    .path("/eureka/v2/apps/{appId}")
+                    .resolveTemplate("appId", APP_NAME)
+                    .request(MediaType.APPLICATION_JSON_TYPE)
+                    .post(json(postBody));
+
+            assertNoContentResponse(response);
+            verify(eurekaServer).registerApplication(isA(InstanceInfo.class));
+        }
+
+        @Test
+        void shouldAttemptToRegisterAndReturn500OnTriggeredFailure() {
```

Comment:
Might want to include an `_` between the `500` and `OnTriggeredFailure` -- I did a double take since I thought it said 5000 🤔

Reply:
Done